lib/ui/page/home/page/chat/widget/AudioPlayer.dart

```diff
@@ -0,0 +1,239 @@
+// Copyright © 2022-2023 IT ENGINEERING MANAGEMENT INC,
+//                       <https://github.com/team113>
+//
+// This program is free software: you can redistribute it and/or modify it under
+// the terms of the GNU Affero General Public License v3.0 as published by the
+// Free Software Foundation, either version 3 of the License, or (at your
+// option) any later version.
+//
+// This program is distributed in the hope that it will be useful, but WITHOUT
+// ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+// FOR A PARTICULAR PURPOSE. See the GNU Affero General Public License v3.0 for
+// more details.
+//
+// You should have received a copy of the GNU Affero General Public License v3.0
+// along with this program. If not, see
+// <https://www.gnu.org/licenses/agpl-3.0.html>.
+
+import 'dart:async';
+import 'dart:typed_data';
+import 'package:flutter/material.dart';
+import 'package:get/get.dart';
+import 'package:path/path.dart';
+import '/config.dart';
+import '/themes.dart';
+import '/ui/page/auth/widget/cupertino_button.dart';
+import '/ui/widget/widget_button.dart';
+import '/util/audio_utils.dart';
+import '/util/message_popup.dart';
+import '/util/platform_utils.dart';
+
+class AudioPlayer extends StatefulWidget {
+
+  /// Optional height this [AudioPlayer] occupies.
+  final double? height;
+
+  /// Optional width this [AudioPlayer] occupies.
+  final double? width;
+
+  /// Callback, called on the video loading errors.
+  final Future<void> Function()? onError;
+
+  /// Constructs a [AudioPlayer] from the provided [url].
+  const AudioPlayer.url(
+      this.url, {
+        super.key,
+        this.checksum,
+        this.height,
+        this.width,
+        this.friendly_name,
+        this.onError,
+      })  : bytes = null,
```

Comment:
```suggestion
      }) : bytes = null,
```

Лишний пробел. `dartfmt` такое бы исправил. Его использование тоже [упомянуто в Contribution Guide](https://github.com/team113/messenger/blob/main/CONTRIBUTING.md#code-style).